packages/mendel-config/src/bundle-config.js

```diff
@@ -7,6 +7,19 @@ function BundleConfig(id, options, config) {
     this.generator = options.generator || 'default';
 
     const {outdir} = config.baseConfig;
+    const supportedOptions = [
+        'entries',
+        'external',
+        'from',
+        'generator',
+        'manifest',
+        'outfile',
+        'outlet',
+        'require',
+        'runtime',
+    ];
```

Comment:
The code you are referring to is done on first level keys. Perhaps we can apply that recursively while plugin specific options can be swept under "options" key that doesnt have any validation or removal.